src/libcore/intrinsics.rs

```diff
@@ -1401,4 +1403,22 @@ extern "rust-intrinsic" {
     /// Emits a `!nontemporal` store according to LLVM (see their docs).
     /// Probably will never become stable.
     pub fn nontemporal_store<T>(ptr: *mut T, val: T);
+
+    /// Initialize the arglist `ap` for processing via `va_arg`.
+    #[cfg(not(stage0))]
+    pub fn va_start(ap: *mut VaListImpl);
+
+    /// Destroy the arglist `ap` after initialization with `va_start` or
+    /// `va_copy`.
+    #[cfg(not(stage0))]
+    pub fn va_end(ap: *mut VaListImpl);
+
+    /// Copy the current location of arglist `ap0` to the arglist `ap1`.
+    #[cfg(not(stage0))]
+    pub fn va_copy(dst: *mut VaListImpl, src: *const VaListImpl);
+
+    /// Loads an argument of type `T` from the `va_list` `ap` and increment the
+    /// argument `ap` points to.
+    #[cfg(not(stage0))]
+    pub fn va_arg<T: Copy>(ap: *mut VaListImpl) -> T;
```

Comment:
My suggestion was to actually use `VaList` here (with `#[repr(transparent)]` on it - you need this *anyway* for FFI correctness, assuming you don't want to expose `VaListImpl` to *all* users) and keep `VaListImpl` private. Then the compiler also doesn't need to extract the `VaListImpl` type.

Reply:
I used the inner value because I didn't want the intrinsics consume the `VaList` and I was unsure of how to define the lifetime of the `VaListImpl`. I'll look into this and try a few things.